src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java

```diff
@@ -509,36 +520,111 @@ private Properties createOverridingProperties() {
 
     /**
      * Returns the list of files (full path name) to process.
-     * @return the list of files included via the filesets.
+     * @return the list of files included via the filesets and source paths.
      */
-    protected List<File> scanFileSets() {
-        final List<File> list = new ArrayList<>();
+    protected List<File> getFilesToCheck() {
+        final List<File> allFiles = new ArrayList<>();
         if (fileName != null) {
             // oops we've got an additional one to process, don't
             // forget it. No sweat, it's fully resolved via the setter.
             log("Adding standalone file for audit", Project.MSG_VERBOSE);
-            list.add(new File(fileName));
+            allFiles.add(new File(fileName));
+        }
+
+        final List<File> filesFromFileSets = scanFileSets();
+        allFiles.addAll(filesFromFileSets);
+
+        final List<File> filesFromSourcePaths = scanSourcePaths();
+        allFiles.addAll(filesFromSourcePaths);
+
+        return allFiles;
+    }
+
+    /**
+     * Retrieves all files from the defined source paths.
+     * @return a list of files defined via source paths.
+     */
+    protected List<File> scanSourcePaths() {
+        final List<File> allFiles = new ArrayList<>();
+
+        for (int i = 0; i < sourcePaths.size(); i++) {
+            final Path sourcePath = sourcePaths.get(i);
+            scanSourcePath(sourcePath, i + 1, allFiles);
         }
+
+        return allFiles;
+    }
+
+    /**
+     * Scans the given source path and adds all files into the given allFiles list.
+     *
+     * @param sourcePath      A source path to scan.
+     * @param sourcePathIndex The index of the given source path. Used in log messages only.
+     * @param allFiles        The resulting list of files.
+     */
+    private void scanSourcePath(Path sourcePath, int sourcePathIndex, List<File> allFiles) {
```

Comment:
No. This is not a good approach. This is an anti-pattern. This method has the side-effect. It modifies the parameter inside. The name says nothing about this fact.